### PR TITLE
Mc mv admin delete dinosaurs

### PR DESCRIPTION
**Pull request by michaelveloso**
Admins can delete any dinosaur, owners can delete only their own dinosaurs

### CODE REVIEW

File: app/controllers/dinosaurs_controller.rb [hunk 1]
```diff
@@ -32,11 +32,20 @@ def create
 
   def edit
     @dinosaur = Dinosaur.find(params[:id])
+    if current_user != @dinosaur.user
+      flash[:errors] = "You can't edit this dinosaur!"
```
Comment (alxjrvs) [inline app/controllers/dinosaurs_controller.rb:36]: Can Admins edit Dinosaurs? This code would suggest not.

File: spec/features/dinosaur/user_can_update_dinosaurs_spec.rb [hunk 2]
```diff
@@ -48,7 +75,7 @@
     end
 
     scenario 'form is pre-filled with current values' do
-      dinosaur = FactoryGirl.create(:dinosaur)
+      dinosaur = Dinosaur.last
```
Comment (alxjrvs) [inline spec/features/dinosaur/user_can_update_dinosaurs_spec.rb]: I would not recommend using `Dinosaur.last`. What made you move away from FactoryGirl?

File: spec/features/user_admin/admin_deletes_dinosaurs_spec.rb [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    visit dinosaur_path(dinosaur)
+
+    click_button "Exctinctify this dinosaur!"
+
+    expect(page).to_not have_content(dinosaur.name)
+  end
+
+  scenario "Dinosaur is no longer in database after deletion" do
+    dinosaur = FactoryGirl.create(:dinosaur)
+    expect(dinosaur).to eq(Dinosaur.last)
```
Comment (alxjrvs) [inline spec/features/user_admin/admin_deletes_dinosaurs_spec.rb]: I don't quite understand your logic here. What is this testing?

File: spec/features/user_admin/admin_deletes_dinosaurs_spec.rb [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+  scenario "Admin sees delete dinosaur button on show page" do
+    dinosaur = FactoryGirl.create(:dinosaur)
+    visit dinosaur_path(dinosaur)
+
+    find_button "Exctinctify this dinosaur!"
+  end
+
+  scenario "After successful deletion, admin is redirected to index" do
+    dinosaur = FactoryGirl.create(:dinosaur)
+    visit dinosaur_path(dinosaur)
```
Comment (alxjrvs) [inline spec/features/user_admin/admin_deletes_dinosaurs_spec.rb]: A lot of this logic could be DRY'd up in a before block

File: app/controllers/dinosaurs_controller.rb (hunk 1 above)
Comment (michaelveloso) [inline app/controllers/dinosaurs_controller.rb:36]: This is actually what we want.  We do not want admins changing peoples words.

File: spec/features/dinosaur/user_can_update_dinosaurs_spec.rb (hunk 2 above)
Comment (michaelveloso) [inline spec/features/dinosaur/user_can_update_dinosaurs_spec.rb]: Fixed. We didn't figure out instance variables in Rspec until a later test.

File: spec/features/user_admin/admin_deletes_dinosaurs_spec.rb (hunk 3 above)
Comment (michaelveloso) [inline spec/features/user_admin/admin_deletes_dinosaurs_spec.rb]: We're making sure the dinosaur no longer shows up in the index page.

Comment (alxjrvs) [inline spec/features/user_admin/admin_deletes_dinosaurs_spec.rb]: I meant specifically with line 59 - Though the merits of testing DB changes in a Feature test are debatable (I go back and forward myself on whether or not it is a good or bad idea), I don't think this line in particular is adding much in the way of value.

File: app/controllers/dinosaurs_controller.rb (hunk 1 above)
Comment (alxjrvs) [inline app/controllers/dinosaurs_controller.rb:36]: Awesome :)

**Pull request merged**